qa-tests-backend/src/test/groovy/ImageScanningTest.groovy

```diff
@@ -767,22 +767,18 @@ class ImageScanningTest extends BaseSpecification {
 
         then:
         "Validate registry based image metadata"
-        def imageDetail = expectedDigestImageFromScan(QUAY_IMAGE_WITH_CLAIR_SCAN_DATA, integrationName)
+        // ROX-23406 - A retry is needed because quay.io will sometimes return
+        // scan details without vulns.
+        withRetry(5, 60) {
+            def imageDetail = expectedDigestImageFromScan(QUAY_IMAGE_WITH_CLAIR_SCAN_DATA, integrationName)
 
-        and:
-        "Validate image scan details"
-        assert imageDetail.scan.dataSource.id != ""
-        assert imageDetail.scan.dataSource.name == scannerName
+            // Validate image scan details
+            assert imageDetail.scan.dataSource.id != ""
+            assert imageDetail.scan.dataSource.name == scannerName
 
-        try {
             assert imageDetail.scan.componentsCount > 0
             assert imageDetail.scan.componentsList.size() > 0
             assert imageDetail.scan.componentsList.vulnsCount.sum { it as Integer } > 0
-        } catch (Exception e) {
-            if (strictIntegrationTesting) {
```

Comment:
This `strictIntegrationTesting` approach to testing is particularly bogus. If a test fails turn it to a skip?%! :upside_down_face:

Reply:
we have more places where we swallow exceptions maybe we should get rid of them